### PR TITLE
ddl,server,store: Fix ineffectual assignment in tests etc.

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

`make check-static` reports errors:

- store/tikv/region_request_test.go:952:19: ineffectual assignment to `err` (ineffassign)
- store/tikv/region_request_test.go:1033:10: ineffectual assignment to `err` (ineffassign)
- ddl/column_type_change_test.go:1775:14: S1039: unnecessary use of fmt.Sprintf (gosimple)
- server/http_handler_test.go:1240:8: ineffectual assignment to `err` (ineffassign)

```
[dvaneeden@dve-carbon tidb]$ make check-static
tools/bin/golangci-lint run -v --disable-all --deadline=3m \
  --enable=misspell \
  --enable=ineffassign \
  --enable=typecheck \
  --enable=varcheck \
  --enable=unused \
  --enable=structcheck \
  --enable=deadcode \
  --enable=gosimple \
  $(go list ./...| grep -vE "cmd|github.com\/pingcap\/tidb\/tests" | sed 's|github.com/pingcap/tidb/||')
INFO [config_reader] Config search paths: [./ /home/dvaneeden/dev/pingcap/tidb/bindinfo /home/dvaneeden/dev/pingcap/tidb /home/dvaneeden/dev/pingcap /home/dvaneeden/dev /home/dvaneeden /home /]
INFO [lintersdb] Active 8 linters: [deadcode gosimple ineffassign misspell structcheck typecheck unused varcheck]
INFO [loader] Go packages loading at mode 575 (compiled_files|exports_file|files|name|types_sizes|deps|imports) took 1.577522455s
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 270.665018ms
INFO [linters context/goanalysis] analyzers took 0s with no stages
INFO [linters context/goanalysis] analyzers took 0s with no stages
INFO [runner/max_same_issues] 1/4 issues with text "ineffectual assignment to `err`" were hidden, use --max-same-issues
INFO [runner] Issues before processing: 48, after processing: 4
INFO [runner] Processors filtering stat (out/in): path_prettifier: 48/48, skip_files: 48/48, skip_dirs: 48/48, nolint: 5/48, max_same_issues: 4/5, cgo: 48/48, autogenerated_exclude: 48/48, source_code: 4/4, path_shortener: 4/4, severity-rules: 4/4, filename_unadjuster: 48/48, exclude: 48/48, exclude-rules: 48/48, diff: 5/5, max_per_file_from_linter: 5/5, max_from_linter: 4/4, identifier_marker: 48/48, path_prefixer: 4/4, sort_results: 4/4, uniq_by_line: 5/5
INFO [runner] processing took 55.84375ms with stages: nolint: 53.957748ms, autogenerated_exclude: 545.947µs, exclude: 501.437µs, identifier_marker: 341.579µs, source_code: 213.322µs, path_prettifier: 206.028µs, skip_dirs: 32.286µs, max_same_issues: 29.107µs, cgo: 5.625µs, filename_unadjuster: 2.924µs, uniq_by_line: 2.912µs, max_from_linter: 1.627µs, path_shortener: 1.277µs, max_per_file_from_linter: 640ns, diff: 428ns, skip_files: 213ns, sort_results: 202ns, severity-rules: 181ns, exclude-rules: 179ns, path_prefixer: 88ns
INFO [runner] linters took 252.977258ms with stages: goanalysis_metalinter: 192.743172ms, unused: 4.334456ms
store/tikv/region_request_test.go:952:19: ineffectual assignment to `err` (ineffassign)
	replicaSelector, err = newReplicaSelector(cache, regionLoc.Region)
	                 ^
store/tikv/region_request_test.go:1033:10: ineffectual assignment to `err` (ineffassign)
	rpcCtx, err = replicaSelector.next(s.bo)
	        ^
ddl/column_type_change_test.go:1775:14: S1039: unnecessary use of fmt.Sprintf (gosimple)
	tk.MustExec(fmt.Sprintf("set time_zone = 'UTC'"))
	            ^
server/http_handler_test.go:1240:8: ineffectual assignment to `err` (ineffassign)
	resp, err = ts.formStatus("/settings", form)
	      ^
INFO File cache stats: 3 entries of total size 211.7KiB
INFO Memory: 23 samples, avg is 114.8MB, max is 205.2MB
INFO Execution took 2.105883648s
make: *** [Makefile:52: check-static] Error 1
```

### Check List 

Tests 

- Unit test

### Release note 

- No release note

### CODE REVIEW

File: server/http_handler_test.go [hunk 1]
```diff
@@ -1237,7 +1237,7 @@ func (ts *HTTPHandlerTestSerialSuite) TestPostSettings(c *C) {
 	}
 	form = make(url.Values)
 	form.Set("tidb_deadlock_history_capacity", "5")
-	resp, err = ts.formStatus("/settings", form)
+	resp, _ = ts.formStatus("/settings", form)
```
Comment (xiongjiwei) [inline server/http_handler_test.go:1240]: we get the error, better to assert it is nil

**Review by xiongjiwei — APPROVED** [review]

**Review by morgo — APPROVED** [review]

**Pull request merged**